pkg/authn/delegating.go

```diff
@@ -1,84 +0,0 @@
-/*
-Copyright 2017 Frederic Branczyk All rights reserved.
-
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
-
-    http://www.apache.org/licenses/LICENSE-2.0
-
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
-
-package authn
-
-import (
-	"errors"
-	"net/http"
-	"time"
-
-	"k8s.io/apiserver/pkg/authentication/authenticator"
-	"k8s.io/apiserver/pkg/authentication/authenticatorfactory"
-	"k8s.io/apiserver/pkg/server/dynamiccertificates"
-	authenticationclient "k8s.io/client-go/kubernetes/typed/authentication/v1"
-)
-
-type DelegatingAuthenticator struct {
-	dynamicClientCA      *dynamiccertificates.DynamicFileCAContent
-	requestAuthenticator authenticator.Request
-}
-
-// NewDelegatingAuthenticator creates an authenticator compatible with the kubelet's needs
-func NewDelegatingAuthenticator(client authenticationclient.TokenReviewInterface, authn *AuthnConfig) (*DelegatingAuthenticator, error) {
-	if client == nil {
-		return nil, errors.New("tokenAccessReview client not provided, cannot use webhook authentication")
-	}
-
-	var (
-		p   *dynamiccertificates.DynamicFileCAContent
-		err error
-	)
-
-	authenticatorConfig := authenticatorfactory.DelegatingAuthenticatorConfig{
-		Anonymous:               false, // always require authentication
-		CacheTTL:                2 * time.Minute,
```

Comment:
What are the k8s.io/apiserver defaults

TODO: post-acceptance, path to make defaults match.

Reply:
Seems like `10 * time.Second`.

https://github.com/kubernetes/kubernetes/blob/75b19b242c34f7c1d1545e156f802f0a06b90533/staging/src/k8s.io/apiserver/pkg/server/options/authentication.go#L207-L220